File: rust_core/hw_keystore/ios/HWKeyStore/HWKeyStore.swift
//
//  HWKeyStore.swift
//  HWKeyStore
//
//  Created by Wallet Developer on 24/02/2023.
//

final class HWKeyStore {
    static let shared = HWKeyStore()

    private let keystore: KeyStore

    private init() {
        self.keystore = KeyStore()

        initHwKeystore(bridge: self.keystore)
    }
}

File: android/app/src/main/kotlin/nl/rijksoverheid/edi/wallet/hwkeystore/HWKeyStore.kt
package nl.rijksoverheid.edi.wallet.hwkeystore

import nl.rijksoverheid.edi.wallet.hwkeystore.bridge.PlatformKeyStore
import uniffi.hw_keystore.KeyStoreBridge
import uniffi.hw_keystore.initHwKeystore

class HWKeyStore {
    companion object {
        val shared = HWKeyStore()

        private val keyStoreBride: KeyStoreBridge = PlatformKeyStore()

        init {
            initHwKeystore(bridge = keyStoreBride)
        }
    }
}
